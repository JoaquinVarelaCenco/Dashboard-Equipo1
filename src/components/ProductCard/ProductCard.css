/* TARJETA */

.main-product__related-product-card {
    display: flex;
    border-radius: 16px;
    background-color: white;
    /* width: 500px; */
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    justify-content: space-between;
  }
  
.main-product__imageAndDesc{
  display: flex;
}

  .section-article__image {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 8px;
  }
  
  .main-product__related-product-card img {
    height: 80px;
    width: 80px;
    border-radius: 16px;
  }
  
  .section-article__desc {
    display: flex;
    flex-direction: column;
    padding: 8px;
    align-self: center;
    gap: 10px;
  }
  
  .desc {
    font-size: 16px;
    font-weight: 700;
    margin-top: 0;
    padding-top: 0;
  }
  
  .desc p {
    margin-top: 0;
    padding-top: 0;
  }
  
  .section-article__price {
    font-size: 16px;
    font-weight: 700;
    line-height: 20px;
    text-align: left;
  }
  
  .section-article__price p {
    margin: 0;
    padding: 0;
  }
  
  .puntos {
    font-size: 8px;
    font-weight: 300;
  }
  
  .button{
  width: 100%;
  }
  

  
  .main-product__related-product {
    display: flex;
    flex-direction: column;
    gap: 10px;
  }
  

  .section-article__goToProduct{
    display: flex;
    align-self: center;
    padding-right: 15px;
  }

  .section-article__goToProduct img{
    width: 40px;
  }
  /* @media (min-width: 785px) {
    .main-product__related-product {
      display: flex;
      flex-direction: row;
    }
    .main-product__related-product__low-quantity{
      display: flex;
      flex-direction: row;
      justify-content: flex-start;
    } */
  /* } */

  /* @media (min-width: 785px) {
    .main-product__related-product-card {
      width: 244px;
      height: 328px;
      flex-direction: column;
      justify-content: space-between;
    }
  
    .section-article__desc {
      flex-direction: row;
      justify-content: space-around;
      padding: 8px;
      align-content: flex-start;
    }
  
    .section-article__price {
      padding-left: 8px;
    }
  
    .section-article__price:first-child {
      display: none;
    }
  
    .desc {
      width: 75%;
      border-right: solid 1px rgba(0, 0, 0, 0.25);
    }
  
    .section-article__image {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: 8px;
    }
  
    .section-article__image img {
      width: 228px;
      height: 240px;
  
      border-radius: 16px;
      margin-left: 8px;
      margin-right: 8px;
    }
  }*/
   