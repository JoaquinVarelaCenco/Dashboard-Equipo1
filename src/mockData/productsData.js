module.exports = [
    {
      "id": 2,
      "title": "iPhone X",
      "description": "SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...",
      "price": 899,
      "rating": { "rate": 4.44, "count": 126 },
      "stock": 34,
      "category": "smartphones",
      "images": [],
      "lastModified": "2022-10-12T01:07:11.630Z"
    },
    {
      "id": 3,
      "title": "Samsung Universe 9",
      "description": "Samsung's new variant which goes beyond Galaxy to the Universe",
      "price": 1249,
      "rating": { "rate": 4.09, "count": 845 },
      "stock": 36,
      "category": "smartphones"
    },
    {
      "id": 23,
      "title": "Orange Essence Food Flavou",
      "description": "Specifications of Orange Essence Food Flavour For Cakes and Baking Food Item",
      "price": 14,
      "rating": { "rate": 4.85, "count": 1645 },
      "stock": 26,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/23/1.jpg",
        "https://dummyjson.com/image/i/products/23/2.jpg",
        "https://dummyjson.com/image/i/products/23/3.jpg",
        "https://dummyjson.com/image/i/products/23/4.jpg",
        "https://dummyjson.com/image/i/products/23/thumbnail.jpg"
      ]
    },
    {
      "id": 24,
      "title": "cereals muesli fruit nuts",
      "description": "original fauji cereal muesli 250gm box pack original fauji cereals muesli fruit nuts flakes breakfast cereal break fast faujicereals cerels cerel foji fouji",
      "price": 46,
      "rating": { "rate": 4.94, "count": 11378 },
      "stock": 113,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/24/1.jpg",
        "https://dummyjson.com/image/i/products/24/2.jpg",
        "https://dummyjson.com/image/i/products/24/3.jpg",
        "https://dummyjson.com/image/i/products/24/4.jpg",
        "https://dummyjson.com/image/i/products/24/thumbnail.jpg"
      ],
      "lastModified": "2022-10-17T14:31:36.884Z"
    },
    {
      "id": 25,
      "title": "Gulab Powder 50 Gram",
      "description": "Dry Rose Flower Powder Gulab Powder 50 Gram â€¢ Treats Wounds",
      "price": 70,
      "rating": { "rate": 4.87, "count": 568 },
      "stock": 47,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/25/1.png",
        "https://dummyjson.com/image/i/products/25/2.jpg",
        "https://dummyjson.com/image/i/products/25/3.png",
        "https://dummyjson.com/image/i/products/25/4.jpg",
        "https://dummyjson.com/image/i/products/25/thumbnail.jpg"
      ]
    }
  ]
  