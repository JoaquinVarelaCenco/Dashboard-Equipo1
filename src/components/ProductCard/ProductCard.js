import "./ProductCard.css";


const ProductCard = ({ title, price, image }) => {
  return (
    <article className="main-product__related-product-card">
      <div className="section-article__image">
        <img
          src={image}
          alt="Imagen de producto"
          className="imagenPrincipal"
        />
      </div>

      <div className="section-article__desc">
        <div className="desc">
          <p>{title}</p>
        </div>

        <div className="section-article__price">
          <p className="valorNumerico">{price}</p>
          <p className="puntos">PUNTOS</p>
        </div>
      </div>
    </article>
  );
};

export default ProductCard;
