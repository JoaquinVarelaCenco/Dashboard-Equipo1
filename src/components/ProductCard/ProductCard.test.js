import { render, screen, act } from "@testing-library/react";
import ProductCard from "./ProductCard";
import { BrowserRouter } from "react-router-dom";
import productsData from "../../mockData/productsData";
import userEvent from "@testing-library/user-event";

describe("Test Tarjeta de producto", () => {
  let data = productsData[0];
  let component;
  beforeEach(async () => {
    component = await render(
      <ProductCard title={data.title} img={data.images[0]} id={data.id} />,
      { wrapper: BrowserRouter }
    );
  });

  test("Tarjeta de productos se debe renderiza correctamente", () => {
    const { container } = component;
    expect(container).toMatchSnapshot();
  });

  test("La tarjeta debe ser un link a la vista de update", () => {
    const linkToUpdate = screen.getByRole("link");
    const expectedLink = "/products/" + data.id;
    expect(linkToUpdate.getAttribute("href")).toBe(expectedLink);
  });

  test("Los valores renderizados en la tarjeta corresponden a las props", () => {
    let title = screen.queryByText(data.title);
    let id = screen.queryByText("#" + data.id);
    expect(title).toBeInTheDocument();
    expect(id).toBeInTheDocument();
  });
});
