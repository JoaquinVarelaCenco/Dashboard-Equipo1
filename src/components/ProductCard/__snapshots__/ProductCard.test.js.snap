// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tarjeta de producto Tarjeta de productos se debe renderiza correctamente 1`] = `
<div>
  <a
    class="product-card-link"
    href="/products/2"
  >
    <article
      class="main-product__related-product-card"
    >
      <div
        class="main-product__imageAndDesc"
      >
        <div
          class="section-article__image"
        >
          <img
            alt="Imagen de producto"
            class="imagenPrincipal hideImage"
            src="no-image.jpeg"
          />
          <div
            class="spinnerWrapper showSpinner"
          >
            <div
              class="spinnerContainer"
            >
              <div
                class="lds-ripple"
              >
                <div />
                <div />
              </div>
            </div>
          </div>
        </div>
        <div
          class="section-article__desc"
        >
          <div
            class="desc"
          >
            <p>
              iPhone X
            </p>
          </div>
          <div
            class="section-article__price"
          >
            <p
              class="valorNumerico"
            >
              #
              2
            </p>
          </div>
        </div>
      </div>
      <div
        class="section-article__goToProduct"
      >
        <img
          alt=""
          src="chevron-right.svg"
        />
      </div>
    </article>
  </a>
</div>
`;
